test/occurrences/double_wrapped.t/run.t

```diff
@@ -0,0 +1,137 @@
+  Main.A.t was used directly 1 times and indirectly 0 times
+  Main.A.x was used directly 1 times and indirectly 0 times
+
+"Aliased" values are not counted since they become persistent
+  $ occurrences_print occurrences-main__C.odoc | sort
+  Main was used directly 0 times and indirectly 2 times
+  Main.A was used directly 1 times and indirectly 1 times
+  Main.A.x was used directly 1 times and indirectly 0 times
+
+Now we can merge all tables
+
+  $ cat > files.map << EOF
+  > occurrences-main__A.odoc
+  > occurrences-main__B.odoc
+  > occurrences-main__C.odoc
+  > EOF
+  $ odoc aggregate-occurrences occurrences-main.odoc occurrences-main__.odoc --file-list files.map -o occurrences-aggregated.odoc
+
+  $ occurrences_print occurrences-aggregated.odoc | sort > all_merged
+  $ cat all_merged
+  Main was used directly 0 times and indirectly 11 times
+  Main.A was used directly 4 times and indirectly 6 times
+  Main.A.(||>) was used directly 1 times and indirectly 0 times
+  Main.A.M was used directly 2 times and indirectly 0 times
+  Main.A.t was used directly 1 times and indirectly 0 times
+  Main.A.x was used directly 2 times and indirectly 0 times
+  Main.B was used directly 1 times and indirectly 0 times
+
+Compare with the one created directly with all occurrences:
+
+  $ odoc count-occurrences -I . -o occurrences-all.odoc
+  $ occurrences_print occurrences-all.odoc | sort > directly_all
+  $ diff all_merged directly_all
+
+We can also include hidden ids:
+
+  $ odoc count-occurrences -I main__B -o occurrences-b.odoc --include-hidden
+  $ occurrences_print occurrences-b.odoc | sort
+  Main was used directly 0 times and indirectly 7 times
+  Main.A was used directly 2 times and indirectly 5 times
+  Main.A.(||>) was used directly 1 times and indirectly 0 times
+  Main.A.M was used directly 2 times and indirectly 0 times
+  Main.A.t was used directly 1 times and indirectly 0 times
+  Main.A.x was used directly 1 times and indirectly 0 times
+  Main__ was used directly 0 times and indirectly 2 times
+  Main__.C was used directly 1 times and indirectly 1 times
+  Main__.C.y was used directly 1 times and indirectly 0 times
+
+  $ odoc count-occurrences -I . -o occurrences-all.odoc --include-hidden
+  $ occurrences_print occurrences-all.odoc | sort
+  Main was used directly 0 times and indirectly 11 times
+  Main.A was used directly 4 times and indirectly 6 times
+  Main.A.(||>) was used directly 1 times and indirectly 0 times
+  Main.A.M was used directly 2 times and indirectly 0 times
+  Main.A.t was used directly 1 times and indirectly 0 times
+  Main.A.x was used directly 2 times and indirectly 0 times
+  Main.B was used directly 1 times and indirectly 0 times
+  Main__ was used directly 0 times and indirectly 2 times
+  Main__.C was used directly 1 times and indirectly 1 times
+  Main__.C.y was used directly 1 times and indirectly 0 times
+  Main__A was used directly 1 times and indirectly 0 times
```

Comment:
`Main__A` is a hidden module.
An alias `Main__.A` is defined in `main__.ml`:
```
module A = Main__A
(** @canonical Main.A *)
```
so, `Main__.A` has `Main.A` as canonical.

However, I don't think that `Main__A` has `Main.A` as canonical: it is aliased with something _which is itself also hidden_. So, it is not "auto-canonicalised", and the `@canonical` tag does not affect it.

One way to fix that would be to say that when we have
```
module X = Y
(** @canonical C *)
```
_and_ `Y` is hidden, then both `X` and `Y` have `C` as canonical. Do you agree? Do you think I should do that in this PR?

Reply:
> and Y is hidden, then both X and Y have C as canonical. Do you agree?

That might already be the case ? The test in `test/xref2/canonical_nested.t/` seems show the same pattern. Perhaps you might need to specially handle `` `Canonical `` identifiers when collecting occurrences ?

> Do you think I should do that in this PR?

No but it would be nice to have a clear picture of the problem.

Reply:
> That might already be the case ?

I tried and I don't think so. Here is my test:

```ocaml
module Cano = struct
  type t
end

module Hidden__one = struct
  type t
end

module Hidden__two = Hidden__one
(** @canonical Test_odoc.A.Cano *)

let y : Hidden__one.t = assert false
let z : Hidden__two.t = assert false
```

In the rendered doc, `z` is shown as having type `Cano.t`, while `y` is shown as having type `{Hidden__one}1.t`.